azure/converters/resourcehealth.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/resourcehealth/mgmt/2020-05-01/resourcehealth"
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+)
+
+// SDKAvailabilityStatusToCondition converts an Azure Resource Health availability status to a status condition.
+func SDKAvailabilityStatusToCondition(availStatus resourcehealth.AvailabilityStatus) *clusterv1.Condition {
+	if availStatus.Properties == nil {
+		return conditions.FalseCondition(infrav1.AzureResourceAvailableCondition, "", "", "")
+	}
+
+	state := availStatus.Properties.AvailabilityState
+
+	if state == resourcehealth.AvailabilityStateValuesAvailable {
+		return conditions.TrueCondition(infrav1.AzureResourceAvailableCondition)
+	}
+
+	var reason string
+	if availStatus.Properties.ReasonType != nil {
+		// CAPI specifies Reason should be CamelCase, though the Azure API
+		// response may include spaces (e.g. "Customer Initiated")
+		reason = strings.ReplaceAll(*availStatus.Properties.ReasonType, " ", "")
```

Comment:
Sounds like we trust that the Azure API will always capitalize the words in the `ReasonType` string, or should we pre-capitalize before we strip out the `" "`?

Reply:
That's all I've seen, but I can't find any docs that describe what the possible values here are, so I think it's best to assume it's basically the wild west. I just added that.